### PR TITLE
Add more AutoAugment policies

### PR DESCRIPTION
**Pull request by stiepan**
## Category:
**New feature** (*non-breaking change which adds functionality*)

## Description:

This PR adds 3 more (apart from the "v0" image-net policy) policies to auto_augment module (reduced image net, reduced cifar-10, svhn). A new function, simply called `auto_augment`, is added to the `auto_augment` module, as a convienience wrapper for applying AA with one of the predefined policies.

The predifined policies as introduced in the AA paper are a bit over-specified: 1. they specify meanigless magnitude bins for augmentations that do not accept magnitudes, 2. they specify some augmentations to be run with 0 probability, 3. they specify some augmentations to be run with magnitude such that the operation is in fact an identity. This PR adds warnings and adjusts the definition of policies to address the points 1. and 2.

Now, all AA/TA and RA modules use both translation_x and translation_y augmentations, so I removed the _get_translation_y helper from auto_aug and moved the _get_translations from RA to common util used across the three modules.

Additonally, the PR fills some gaps in the documentation (i.e. max_translation_abs/rel) and the docs in utilities.

### Tests:

- [ ] Existing tests apply
- [x] New tests added
  - [x] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [x] Documentation updated
  - [x] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: DALI-3299

### CODE REVIEW

File: dali/test/python/auto_aug/test_auto_augment.py [hunk 1]
```diff
@@ -91,57 +99,6 @@ def pipeline():
...
-    for i, sample in enumerate(output):
-        sample = np.array(sample)
-        if offset_fraction == 1 and extent != "width":
-            assert np.all(sample == fill_value), f"sample_idx: {i}"
-        else:
-            background_count = np.sum(sample == fill_value)
-            assert background_count / sample.size < 0.1, \
-                f"sample_idx: {i}, {background_count / sample.size}"
-
-
```
Comment (stiepan) [inline dali/test/python/auto_aug/test_auto_augment.py:144]: It's now common utility for AA/TA/RA modules. Tested in `test_augmentations`.

File: dali/test/python/auto_aug/test_rand_augment.py [hunk 2]
```diff
@@ -66,61 +66,6 @@ def pipeline():
...
-    for i, sample in enumerate(output):
-        sample = np.array(sample)
-        if offset_fraction == 1:
-            assert np.all(sample == fill_value), f"sample_idx: {i}"
-        else:
-            background_count = np.sum(sample == fill_value)
-            assert background_count / sample.size < 0.1, \
-                f"sample_idx: {i}, {background_count / sample.size}"
-
-
```
Comment (stiepan) [inline dali/test/python/auto_aug/test_rand_augment.py:123]: It's now common utility for AA/TA/RA modules. Tested in `test_augmentations`.

File: dali/test/python/auto_aug/test_trivial_augment.py [hunk 3]
```diff
@@ -63,54 +63,6 @@ def pipeline():
...
-            all_black += np.all(sample == fill_value)
-        assert all_black
-    else:
-        for i, sample in enumerate(output):
-            sample = np.array(sample)
-            background_count = np.sum(sample == fill_value)
-            assert background_count / sample.size < 0.1, \
-                f"sample_idx: {i}, {background_count / sample.size}"
-
-
```
Comment (stiepan) [inline dali/test/python/auto_aug/test_trivial_augment.py:113]: It's now common utility for AA/TA/RA modules. Tested in `test_augmentations`.

File: dali/python/nvidia/dali/auto_aug/rand_augment.py [hunk 4]
```diff
@@ -237,20 +251,3 @@ def get_rand_augment_non_monotonic_suite(
...
-    if use_shape:
-        return [
-            a.translate_x.augmentation((0, max_translate_width), True),
-            a.translate_y.augmentation((0, max_translate_height), True),
-        ]
-    else:
-        return [
-            a.translate_x_no_shape.augmentation((0, max_translate_width), True),
-            a.translate_y_no_shape.augmentation((0, max_translate_height), True),
-        ]
```
Comment (stiepan) [inline dali/python/nvidia/dali/auto_aug/rand_augment.py:256]: It's now common utility for AA/TA/RA modules. Tested in `test_augmentations`.

File: dali/python/nvidia/dali/auto_aug/trivial_augment.py [hunk 5]
```diff
@@ -176,20 +189,3 @@ def get_trivial_augment_wide_suite(
...
-    if use_shape:
-        return [
-            a.translate_x.augmentation((0, max_translate_width), True),
-            a.translate_y.augmentation((0, max_translate_height), True),
-        ]
-    else:
-        return [
-            a.translate_x_no_shape.augmentation((0, max_translate_width), True),
-            a.translate_y_no_shape.augmentation((0, max_translate_height), True),
-        ]
```
Comment (stiepan) [inline dali/python/nvidia/dali/auto_aug/trivial_augment.py:195]: It's now common utility for AA/TA/RA modules. Defined in `core._utils`.

File: dali/test/python/auto_aug/test_augmentations.py [hunk 6]
```diff
@@ -375,3 +377,65 @@ def get_batch():
...
+    assert extent in ('height', 'width', 'both', 'none'), f"{extent}"
+    if extent != 'none':
+        if use_shape:
+            param_shape = [1., 1.]
+            param_name = "max_translate_rel"
+        else:
+            param_shape = shape
+            param_name = "max_translate_abs"
+        if extent == 'both':
+            param = [param_shape[0] * offset_fraction, param_shape[1] * offset_fraction]
```
Comment (jantonguirao) [inline dali/test/python/auto_aug/test_augmentations.py:400]: nitpick:
```suggestion
            param = offset_fraction * param_shape[:2]
```
or
```suggestion
            param = offset_fraction * param_shape
```

**Review by jantonguirao — APPROVED** [review]

**Comment by stiepan** [comment]
Rebased onto https://github.com/NVIDIA/DALI/pull/4747

**Comment by stiepan** [comment]
!build

**Comment by stiepan** [comment]
Rebased onto https://github.com/NVIDIA/DALI/pull/4751

**Comment by stiepan** [comment]
!build

**Review by klecki — APPROVED** [review]

**Comment by stiepan** [comment]
!build

**Pull request merged**